examples/dsa_hisa/hisa.py

```
@@ -0,0 +1,240 @@
+        block_sparse_logits,
+        k=topk_tokens_eff,
+        dim=-1,
+    ).indices  # [M, topk_tokens_eff] int64
+
+    # ------------------------------------------------------------------
+    # Stage 3 (post, Python): translate slot ids → absolute K token
+    # position → per-query relative offset (matches vLLM indexer
+    # output buffer). Slots whose relative offset falls outside the
+    # query's visible range are set to -1.
+    # ------------------------------------------------------------------
+    # slot = block_id_in_topk × k_block_size + offset_in_block
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)